### ISSUE TITLE
compilation error

### ISSUE THREAD

**Issue by saker88** [vekexasia/comfoair-esp32#78]
Im getting error and I do not know how to fix this ?

[away_indicator] is an invalid option for [comfoair]. Please check the indentation.

ive tried to rewrite that part , space check and nothing seems to help.. 

please help :D im blind and lost in fog here.

**Comment by vekexasia** [comment]
Hey @saker88 can u share ur config please?

**Comment by saker88** [comment]
esp config

`esphome:
  name: comfoair
  friendly_name: comfoair

esp32:
  board: esp32dev
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: "6xoH9BDiFKQT+D9M4PfCufjaH0M7duaJOKlVMqiYlJM="

ota:
  - platform: esphome
    password: "71fcd5be07c55b0dfe28e97cf74d42bb"

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Comfoair Fallback Hotspot"
    password: "GymQmucjlJg9"

captive_portal:

external_components:
  - source: github://vekexasia/comfoair-esp32
    components: [ comfoair ]

comfoair:
  id: comfo
  rx_pin: GPIO5
  tx_pin: GPIO4
  supply_fan_speed:
    name: supply_fan_speed
    disabled_by_default: false
    force_update: false
    unit_of_measurement: rpm
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 100.0
  fan_speed:
    name: fan_speed
    disabled_by_default: false
    force_update: false
    unit_of_measurement: ''
    accuracy_decimals: 0
    filters:
      - throttle: 300s
  next_fan_change:
    name: next_fan_change
    disabled_by_default: false
    force_update: false
    unit_of_measurement: s
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 180.0
  next_bypass_change:
    name: next_bypass_change
    disabled_by_default: false
    force_update: false
    unit_of_measurement: s
    accuracy_decimals: 0
  exhaust_fan_duty:
    name: exhaust_fan_duty
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  supply_fan_duty:
    name: supply_fan_duty
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  exhaust_fan_flow:
    name: exhaust_fan_flow
    disabled_by_default: false
    force_update: false
    unit_of_measurement: m³/h
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 5.0
  supply_fan_flow:
    name: supply_fan_flow
    disabled_by_default: false
    force_update: false
    unit_of_measurement: m³/h
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 5.0
  exhaust_fan_speed:
    name: exhaust_fan_speed
    disabled_by_default: false
    force_update: false
    unit_of_measurement: rpm
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 100.0
  power_consumption_current:
    name: power_consumption_current
    disabled_by_default: false
    force_update: false
    unit_of_measurement: W
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 15.0
  energy_consumption_ytd:
    name: energy_consumption_ytd
    disabled_by_default: false
    force_update: false
    unit_of_measurement: kWh
    accuracy_decimals: 0
  energy_consumption_since_start:
    name: energy_consumption_since_start
    disabled_by_default: false
    force_update: false
    unit_of_measurement: kWh
    accuracy_decimals: 0
  days_remaining_filter:
    name: days_remaining_filter
    disabled_by_default: false
    force_update: false
    unit_of_measurement: days
    accuracy_decimals: 0
  avoided_heating_actual:
    name: avoided_heating_actual
    disabled_by_default: false
    force_update: false
    unit_of_measurement: W
    accuracy_decimals: 2
    filters:
      - throttle_average: 300s
      - filter_out: nan
  avoided_heating_ytd:
    name: avoided_heating_ytd
    disabled_by_default: false
    force_update: false
    unit_of_measurement: Wh
    accuracy_decimals: 0
  avoided_heating_total:
    name: avoided_heating_total
    disabled_by_default: false
    force_update: false
    unit_of_measurement: Wh
    accuracy_decimals: 0
  avoided_cooling_actual:
    name: avoided_cooling_actual
    disabled_by_default: false
    force_update: false
    unit_of_measurement: W
    accuracy_decimals: 2
    filters:
      - or:
        - throttle: 300s
        - delta: 1.0
  avoided_cooling_ytd:
    name: avoided_cooling_ytd
    disabled_by_default: false
    force_update: false
    unit_of_measurement: Wh
    accuracy_decimals: 0
  avoided_cooling_total:
    name: avoided_cooling_total
    disabled_by_default: false
    force_update: false
    unit_of_measurement: Wh
    accuracy_decimals: 0
  bypass_state:
    name: bypass_state
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
  rmot:
    name: rmot
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
  target_temp:
    name: target_temp
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
  pre_heater_temp_before:
    name: pre_heater_temp_before
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  post_heater_temp_before:
    name: post_heater_temp_before
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    filters:
      - or:
        - throttle: 300s
        - delta: 0.3
  extract_air_temp:
    name: extract_air_temp
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
  exhaust_air_temp:
    name: exhaust_air_temp
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    filters:
      - or:
        - throttle: 300s
        - delta: 0.3
  outdoor_air_temp:
    name: outdoor_air_temp
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    filters:
      - or:
        - throttle: 300s
        - delta: 0.3  
  pre_heater_temp_after:
    name: pre_heater_temp_after
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    device_class: temperature
    filters:
      - or:
        - throttle: 300s
        - delta: 0.3
  post_heater_temp_after:
    name: post_heater_temp_after
    disabled_by_default: false
    force_update: false
    unit_of_measurement: °C
    accuracy_decimals: 1
    filters:
      - or:
        - throttle: 300s
        - delta: 0.3
  extract_air_humidity:
    name: extract_air_humidity
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  exhaust_air_humidity:
    name: exhaust_air_humidity
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  outdoor_air_humidity:
    name: outdoor_air_humidity
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  pre_heater_hum_after:
    name: pre_heater_hum_after
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2
  supply_air_humidity:
    name: supply_air_humidity
    disabled_by_default: false
    force_update: false
    unit_of_measurement: '%'
    accuracy_decimals: 0
    filters:
      - or:
        - throttle: 300s
        - delta: 2.0
  away_indicator:
    name: away_indicator
    disabled_by_default: false
  operating_mode:
    name: operating_mode
    disabled_by_default: false
  bypass_activation_mode:
    name: bypass_activation_mode
    disabled_by_default: false
  temp_profile:
    name: temp_profile
    disabled_by_default: false

# see: https://esphome.io/components/time.html
time:
  - platform: homeassistant
    id: homeassistant_time

# Enable Web server
web_server:
  port: 80

button:
  - platform: template
    name: "Boost 10m"
    id: comfoair_boost_10min
    on_press:
      then:
        - lambda: |- 
            id(comfo).send_command("boost_10_min");

select:
  - platform: template
    name: "Belüftungsstärke"
    id: comfoair_ventilation_level
    optimistic: true
    options:
      - "Abwesend"
      - "Stufe 1"
      - "Stufe 2"
      - "Stufe 3"
    set_action:
      - lambda: |- 
          if (x == "Stufe 1") {
            id(comfo).send_command("ventilation_level_1");
          }
          if (x == "Stufe 2") {
            id(comfo).send_command("ventilation_level_2");
          }
          if (x == "Stufe 3") {
            id(comfo).send_command("ventilation_level_3");
          }
          if (x == "Abwesend") {
            id(comfo).send_command("ventilation_level_0");
          }
  - platform: template
    name: "Mode"
    id: comfoair_mode
    optimistic: true
    options:
      - "auto"
      - "manual"
    set_action:
      - lambda: |- 
          if (x == "auto") {
            id(comfo).send_command("auto");
          }
          if (x == "manual") {
            id(comfo).send_command("manual");
          }`

**Comment by vekexasia** [comment]
hey can u check this comment here ?  https://github.com/vekexasia/comfoair-esp32/issues/76#issuecomment-2213244441

you do not really need all the other configuration elements now you can specify a global filter like so:

```yaml
comfoair:
  global_filters:
    - throttle: 10s
```

**Comment by mattapo** [comment]
> Im getting error and I do not know how to fix this ?
> 
> [away_indicator] is an invalid option for [comfoair]. Please check the indentation.
> 
> ive tried to rewrite that part , space check and nothing seems to help..
> 
> please help :D im blind and lost in fog here.

"away_indicator" is just "away". Edit the yaml file and the error will not appear

**Comment by vekexasia** [comment]
hey @mattapo  i already removed all the entries from the sample most people dont need throttling per single sensor https://github.com/vekexasia/comfoair-esp32/blob/8d63d76fc53c933127928f9cec91cd4bae2cf187/docs/esphome_sample.yaml#L5C1-L8C16

**Comment by saker88** [comment]
Hey, I managed to solve my issue by omitting the away_indicator and rewriting whole code once more. Now, I need to handle the wiring. I've noticed that you're using the M5Stack Atom (I assume) and in your code, you're referring to GPIO5 and GPIO4. On the PCB, you're connecting to pins G21 and G25. How do these correlate with the ESP32 WROOM dev board? Are they the same, or do I need to redefine them to different pins?

Thanks for updating the sample code and explaining the basics. I'm new to the HA system and YAML, and I'm trying to learn and understand by running shared projects. I appreciate your patience with us "noobs." It's great that even seemingly silly problems or questions don't go unanswered and are addressed in a professional manner.
